tests/core/math/test_projection.h

```diff
@@ -0,0 +1,637 @@
+			Vector4(-2.16, 70.96, -59.63, 23.87))
+							.get_fov();
+
+	CHECK(expected == doctest::Approx(actual));
+}
+
+TEST_CASE("[Projection] get_fovy") {
+	real_t expected = 1.32997;
+	real_t actual = Projection().get_fovy(1.0, 1.33);
+
+	CHECK(expected == doctest::Approx(actual));
+}
+
+TEST_CASE("[Projection] get_lod_multiplier") {
+	float expected = 1;
+	float actual = Projection().get_lod_multiplier();
+
+	CHECK(expected == doctest::Approx(actual));
+}
+
+TEST_CASE("[Projection] get_pixels_per_meter") {
+	float expected = 55;
+	float actual = Projection(
+			Vector4(-29.32, 43.39, -94.85, -89.95),
+			Vector4(55.01, -55.50, 87.92, 36.67),
+			Vector4(48.63, 19.80, 75.48, 44.12),
+			Vector4(-2.16, 70.96, -59.63, 23.87))
+						   .get_pixels_per_meter(64);
+
+	CHECK(expected == doctest::Approx(actual));
+}
+
+TEST_CASE("[Projection] get_projection_plane") {
+	SUBCASE("PLANE_NEAR") {
+		Plane expected = Plane(Vector3(0, 0, -1), 1);
+		Plane actual = Projection().get_projection_plane(Projection::Planes::PLANE_NEAR);
+
+		CHECK(expected == actual);
+	}
+
+	SUBCASE("PLANE_FAR") {
+		Plane expected = Plane(Vector3(0, 0, 1), 1);
+		Plane actual = Projection().get_projection_plane(Projection::Planes::PLANE_FAR);
+
+		CHECK(expected == actual);
+	}
+
+	SUBCASE("PLANE_LEFT") {
+		Plane expected = Plane(Vector3(-1, 0, 0), 1);
+		Plane actual = Projection().get_projection_plane(Projection::Planes::PLANE_LEFT);
+
+		CHECK(expected == actual);
+	}
+
+	SUBCASE("PLANE_TOP") {
+		Plane expected = Plane(Vector3(0, 1, 0), 1);
+		Plane actual = Projection().get_projection_plane(Projection::Planes::PLANE_TOP);
+
+		CHECK(expected == actual);
+	}
+
+	SUBCASE("PLANE_RIGHT") {
+		Plane expected = Plane(Vector3(1, 0, 0), 1);
+		Plane actual = Projection().get_projection_plane(Projection::Planes::PLANE_RIGHT);
+
+		CHECK(expected == actual);
+	}
+
+	SUBCASE("PLANE_BOTTOM") {
+		Plane expected = Plane(Vector3(0, -1, 0), 1);
+		Plane actual = Projection().get_projection_plane(Projection::Planes::PLANE_BOTTOM);
+
+		CHECK(expected == actual);
+	}
+}
+
+TEST_CASE("[Projection] get_viewport_half_extents ") {
+	SUBCASE("get_viewport_half_extents of a generic matrix") {
+		Vector2 expected(0.189962f, 0.867947f);
+		Vector2 actual = Projection(
+				Vector4(-29.32, 43.39, -94.85, -89.95),
+				Vector4(55.01, -55.50, 87.92, 36.67),
+				Vector4(48.63, 19.80, 75.48, 44.12),
+				Vector4(-2.16, 70.96, -59.63, 23.87))
+								 .get_viewport_half_extents();
+
+		CHECK(expected.x == doctest::Approx(actual.x));
+		CHECK(expected.y == doctest::Approx(actual.y));
+	}
+
+	SUBCASE("get_viewport_half_extents of null matrix") {
+		Vector2 expected(0.0, 0.0);
+		Vector2 actual = Projection(
+				Vector4(0, 0, 0, 0),
+				Vector4(0, 0, 0, 0),
+				Vector4(0, 0, 0, 0),
+				Vector4(0, 0, 0, 0))
+								 .get_viewport_half_extents();
+
+		CHECK(expected.is_equal_approx(actual));
+	}
+}
+
+TEST_CASE("[Projection] get_z_far") {
+	float expected = 1;
+	float actual = Projection().get_z_far();
```

Comment:
```suggestion
	real_t actual = Projection().get_z_far();
```
Use the precision based value for these, look at the return value of the methods